python/astshim/fitsChanContinued.py

```diff
@@ -0,0 +1,378 @@
+    or the supplied integer is acceptable to the FitsChan.
+    """
+    if isinstance(name, int):
+        # index will be zero-based
+        if name >= 0 and name < self.nCard:
+            return True
+    elif isinstance(name, str):
+        currentCard = self.getCard()
+        try:
+            self.clearCard()
+            result = self.findFits(name, False)
+        finally:
+            self.setCard(currentCard)
+        if result.found:
+            return True
+
+    return False
+
+
+FitsChan.__contains__ = contains
+
+
+def getitem(self, name):
+    """Return a value associated with the supplied name.
+
+    Parameters
+    ----------
+    name : `str` or `int`
+        If the FitsChan is being accessed by integer index the returned value
+        will be the corresponding 80-character card.  Index values are 0-based.
+        A negative index counts from the end of the FitsChan.
+        If the FitsChan is being accessed by string the returned value will
+        be the scalar value associated with the first card that matches the
+        supplied name.
+
+    Raises
+    ------
+    IndexError
+        Raised if an integer index is provided and the index is out of range.
+    KeyError
+        Raised if a string is provided and that string is not present in
+        the FitsChan. Also raised if the supplied name is neither an integer
+        not a string.
+    RuntimeError
+        Raised if there is some problem accessing the value in the FitsChan.
+    """
+
+    # Save current card position
+    currentCard = self.getCard()
+
+    if isinstance(name, int):
+        # Calculate position in FitsChan (0-based to 1-based)
+        newpos = _calc_card_pos(self, name)
+        self.setCard(newpos)
+        try:
+            result = self.findFits("%f", False)
+        finally:
+            self.setCard(currentCard)
+        if not result.found:
+            raise IndexError(f"No FITS card at index {name}")
+        return result.value
+
+    elif isinstance(name, str):
+
+        try:
+            # Rewind FitsChan so we search all cards
+            self.clearCard()
+
+            # We are only interested in the first matching card
+            result = self.findFits(name, False)
+            if not result.found:
+                raise KeyError(f"{name}'")
+
+            this_name, value = _get_current_card_value(self)
+            if this_name != name:
+                raise RuntimeError(f"Internal inconsistency in get: {this_name} != {name}")
+
+        finally:
+            # Reinstate the original card position
+            self.setCard(currentCard)
+
+        return value
+
+    raise KeyError(f"Supplied key, '{name}' of unsupported type")
+
+
+FitsChan.__getitem__ = getitem
+
+
+def setitem(self, name, value):
+    """Set a new value.
+
+    Parameters
+    ----------
+    name : `str` or `int`
+        If the name is an integer index the returned value is the corresponding
```

Comment:
"returned value"? I think you have a bit of a copy-paste issue here. Most of this text is fine but there are a few bits that need cleanup for this being getitem instead of setitem. In particular it still does not say (clearly enough) that when name is an integer then the value must be an 80 character formatted card.

Which raises a question I forgot to ask earlier: what happens if `name` is an integer and `value` is **not** a string, or is a string but is not 80 characters long? I hope you'd raise an exception.